### PR TITLE
2021 07 pubmed links

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
PubMed links in TextView component. [TRM-24837](https://www.ebi.ac.uk/panda/jira/browse/TRM-24837)

## Approach
 - Split text by PubMed IDs, replace ID with link
 - Add line break at the end of a sentence with PubMed links
 - Replaced section with div (cards were already a section, there's no need to have further sections within the cards if it's just new paragraphs)
   - By the way, couldn't use `<p>` for paragraph because of the embedded evidence tags that contain divs
 - Replaced headings that didn't have the correct heading level in entry pages
 - ~Some small prettier changes got in the commits~ (fixed upstream)

## Testing
Updated mock, test, and snapshot, to cover more cases

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
 describe('FreeText component', () => {
   test('should render free text CC', () => {
-    const { asFragment } = render(<FreeTextView comments={FreeTextUIData} />);
+    const { asFragment } = customRender(
+      <FreeTextView comments={freeTextUIData} />
+    );
     expect(asFragment()).toMatchSnapshot();
+
+    // PubMed links
+    expect(screen.getAllByRole('link')).toHaveLength(2);
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx:16]: I'm also wondering whether it would be worth testing edge cases like comment with no PubMed id to prevent accidental regression.

**Review by xwatkins — CHANGES_REQUESTED** [review]

File: src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aurel-l) [reply src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx:16]: I'm assuming you wanted more combinations of the test, rather than splitting this specific assertion into 3, right?
I ended up making 3 tests, with similar assertions, but different props (plain text+evidence tags / text with inline pubmeds+evidence tags / plain text+no evidence tag).

[the remark this answers is not in the document]
Comment (xwatkins) [reply src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx:31]: With Jest I always find it weird to have multiple assertions in the same test - I know it's different from unit tests, and I guess as long as you're testing a small workflow it's ok. In this case though, I think 2 of the assertions were already covered by the snapshot?

**Review by xwatkins — APPROVED** [review]

File: src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aurel-l) [reply src/uniprotkb/components/protein-data-views/__tests__/FreeTextView.spec.tsx:31]: Ah, I see, I'll split them into multiple suites then, for each combination of prop, and then test the things in different tests.
I'd be keen on removing the snapshots here, as I'm already asserting on the important bit which are the links and evidence tags, what do you think?
